src/seedu/addressbook/data/person/Name.java

```diff
@@ -55,6 +56,37 @@ public boolean equals(Object other) {
                 || (other instanceof Name // instanceof handles nulls
                 && this.fullName.equals(((Name) other).fullName)); // state check
     }
+    
+    public boolean isSimilar(Name other){
+    	String myName = this.fullName.toLowerCase();
+    	String theirName = other.fullName.toLowerCase();
+    	
+    	if (myName == null || theirName == null)
+    		return false;
+    	
```

Comment:
Code should be written to make failing tests pass, but your tests don't have the case where a name is null. Make sure you write the test then the code!